### PR TITLE
feat(users): Add UserSearch component

### PR DESCRIPTION
**Pull request by damassi**
This adds a new AutocompleteInput to the /users page, as a quick way to get to a user when you don't know their internalID. Takes advantage of our new `<LazyQueryRenderer />` component, too. 

email-input

Also updates `LazyQueryRenderer` with a couple more props:
- `enabled: boolean`: only perform fetch if certain conditions are met 
- `fetchPolicy`: allows you to update the relay fetch policy for the query

### CODE REVIEW

File: src/pages/users/components/UserSearch.tsx [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+        query UserSearchQuery($email: String!) {
+          user(email: $email) {
+            name
+            internalID
+          }
+        }
+      `}
+      variables={{ email }}
+      enabled={/\S+@\S+\.\S+/.test(email)} // only fetch if email
+      render={({ isLoading, data }) => {
```
Comment (ovasdi) [inline src/pages/users/components/UserSearch.tsx:24]: **Question**: so this `render` is a callback that executes when `LeazyQueryRenderer` returns a fragment with two props, `isLoading` and `data`, those are then passed down to `AutocompleteInput` which is what ends up getting rendered by this component. Is that correct?

**Review by ovasdi — APPROVED** [review]
Great stuff. ✨

**Comment by mc-jones** [comment]
Thanks for this, @damassi! 💯 🙏

File: src/pages/users/components/UserSearch.tsx (hunk 1 above)
Comment (damassi) [reply src/pages/users/components/UserSearch.tsx:24]: Yup, if you're rendering `LazyQueryRenderer`, `render` always gets called and attempts to render the component whether `isLoading` is true or false. What this does in addition tho is to report `isLoading` status -- which you can't typically get in a code-usable way via react suspense.

**Pull request merged**